### PR TITLE
desktop: browser view and custom title bar (experimental)

### PR DESCRIPTION
**Pull request by nsakaimbo**
### Description

Experimenting with customizing the Desktop app's title bar. At minimum, this would allow us to enable "Back", "Forward" and "Home" navigation buttons. Further looking, we might be able to use these basic concepts to do even fancier things like enabling multi-account support via a sidebar switcher.

Excuse my poor CSS and HTML skills - and and all changes welcome.

<p align="center">
Screen Shot 2021-05-24 at 8 31 13 PM
</p>

### Context

Not only is `BrowserView` a recommended Electron best practice, but it allows us to remote load Calypso/WordPress.com in a nested "content view". This allows us to customize the "chrome" of the application (i.e the application's native title bar). Specifically, we would like to provide users a better in-app experience with the re-architected WP-Desktop app such that they don't need to be bounded out of the application when navigating to Atomic/self-hosted sites, as well as accessing items such as full-screen images, etc. Adding always-persistent back/forward/home navigation elements allows users to navigate more freely and always have the ability to return to their WordPress.com landing page, regardless of navigation context.

This approach was heavily inspired by the tryshift application ([link](https://tryshift.com/)).

### Overview: BrowserView and BrowserWindow

TL;DR: the main window object (`BrowserWindow` instance) encapsulates embedded content view a `BrowserView` instance. We load custom navigation elements via local HTML, and load Calypso/WordPress.com into the `BrowserView`. The `BrowserView` instance is very similar to the `BrowserWindow` and allows us to use virtually all of the existing `BrowserWindow` APIs via its `webContents` object.

### TODOs

- [x] Hook up "Back" and "Home" buttons ("Forward" button is probably unnecessary - we can probably remove this?)
- [x] Devtools: open dev tools in detached mode (using `BrowserView` makes it so the main window's dev tools are hidden behind the embedded `BrowserView` instance)
- [x] Test on Windows
- [x] Test on Linux
- [ ] Verify authentication continues to work as expected. Not sure if related but it seems using `BrowserView` results in being prompted with the `Log into your account -> Continue` flow (this is the ["continue as user"](https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/login/continue-as-user.jsx) flow in Calypso).

### CODE REVIEW

File: desktop/public_desktop/index.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+<html>
+	<head>
+		<meta charset="UTF-8" />
+		<link rel="stylesheet" href="default.css" />
+	</head>
+	<body style="background: #24282d; color: #24282d">
+		<div id="title-bar">
+			<div id="title-bar-buttons">
+				<div class="nav-button" id="nav-button-back" onclick="navButtonClicked(this.id)">
+						<svg class="nav-button-icon"
```
Comment (belcherj) [inline desktop/public_desktop/index.html]: Indentation is off here.

File: desktop/public_desktop/wordpress.svg [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+            </g>
+        </g>
+    </g>
+</svg>
```
Comment (belcherj) [inline desktop/public_desktop/wordpress.svg]: Needs a new line

**Review by belcherj — CHANGES_REQUESTED** [review]

**Review by belcherj — APPROVED** [review]
Ship it.
